src/libraries/Microsoft.PowerFx.Connectors/ODataVisitor.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Specialized;
+using System.Globalization;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.IR.Symbols;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    internal class ODataVisitor : IRNodeVisitor<string, object>
+    {
+        internal static readonly ODataVisitor I = new ODataVisitor();
+
+        public override string Visit(TextLiteralNode node, object context)
+        {
+            return $"'{node.LiteralValue}'";
+        }
+
+        public override string Visit(NumberLiteralNode node, object context)
+        {
+            return node.LiteralValue.ToString(CultureInfo.InvariantCulture);
+        }
+
+        public override string Visit(BooleanLiteralNode node, object context)
+        {
+            return node.LiteralValue.ToString(CultureInfo.InvariantCulture);
+        }
+
+        public override string Visit(ColorLiteralNode node, object context)
+        {
+            throw new System.NotImplementedException();
```

Comment:
We may want a specific `NonDelegableException` that we can catch and fall back to trying in-memory.

Reply:
Added